cmd/root.go

```diff
@@ -111,22 +95,49 @@ var (
 				log.Fatal(err)
 			}
 		},
-	}, &algodEndpoint, &algodToken)
+	}, &algodData)
 )
 
-// init initializes the application, setting up logging, commands, and version information.
-func init() {
-	log.SetReportTimestamp(false)
+// NeedsToBeRunning ensures the Algod software is installed and running before executing the associated Cobra command.
+func NeedsToBeRunning(cmd *cobra.Command, args []string) {
+	if force {
+		return
+	}
+	if !algod.IsInstalled() {
+		log.Fatal(explanations.NotInstalledErrorMsg)
+	}
+	if !algod.IsRunning() {
+		log.Fatal(explanations.NotRunningErrorMsg)
+	}
+}
 
-	// Configure Version
-	if Version == "" {
-		Version = "unknown (built from source)"
+// NeedsToBeStopped ensures the operation halts if Algod is not installed or is currently running, unless forced.
+func NeedsToBeStopped(cmd *cobra.Command, args []string) {
+	if force {
+		return
 	}
-	rootCmd.Version = Version
+	if !algod.IsInstalled() {
+		log.Fatal(explanations.NotInstalledErrorMsg)
+	}
+	if algod.IsRunning() {
+		log.Fatal(explanations.RunningErrorMsg)
+	}
+}
 
+// init initializes the application, setting up logging, commands, and version information.
+func init() {
+	log.SetReportTimestamp(false)
 	// Add Commands
 	if runtime.GOOS != "windows" {
```

Comment:
What would this result in under windows? Do we / should we error out with a message about using through WSL?

Reply:
Depends on which binary they are running in WSL. The linux binary will only work in WSL so we won't be able to expose the error if they run in it Windows. We are not shipping a windows binary which would be able to alert the user in this context.

Between not having the command available and forcing users to manually build the project that should be enough. If we error on Windows we are effectively forcing users to fork the project to make code changes.